js/hw_3_js_react_1/src/components/Board/Board.jsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useEffect } from "react";
+
+import { generateNewFood } from "../../function/Food/generateNewFood";
+import { growSnake } from "../../function/Snake/growSnake";
+import { snakeBodyWithTail } from "../../function/Snake/snakeBodyWithTail";
+import {
+  getIsSnakeHeadOutsideBoard,
+  getIsSnakeHeadOnBody,
+  getIsTimeOut,
+} from "../../function/GameOver/gameOver";
+import { getHasEatenFood } from "../../function/Snake/getHasEatenFood";
+import { GameTimer } from "../GameTimer/GameTimer";
+import { GameScore } from "../GameScore/GameScore"
+
+import "./Board.css";
+
+const boardWidth = 25;
+const board = Array(boardWidth).fill(Array(boardWidth).fill(0));
+const initialSnakeBody = [
+  [10, 10],
+  [10, 11],
+  [10, 12],
+  [10, 13],
+  [10, 14],
+];
+const initialFood = [[]];
+const speed = 50;
+const initialTime = 120
+
+const Board = () => {
+  const [snake, setSnake] = useState([initialSnakeBody]);
+  const [food, setFood] = useState(initialFood);
+  const [direction, setDirection] = useState("");
+  const [gameOver, setGameOver] = useState(false);
+  const [score, setScore] = useState(0);
+  const [time, setTime] = useState(initialTime);
+  const [start, setStart] = useState(false);
+  const [pause, setPause] = useState(false);
+
+  useEffect(() => {
+    if (gameOver) {
+      return;
+    }
+    document.addEventListener("keydown", changeDirection);
+    return () => document.removeEventListener("keydown", changeDirection);
+  });
+
+  useEffect(() => {
+    if (gameOver) {
+      setStart(false);
+    }
+    if (start && !pause) {
+      const intervalId = setInterval(() => {
+        setTime((prevTime) => prevTime - 1);
+      }, 1000);
+      return () => clearInterval(intervalId);
+    }
+  }, [start, time, gameOver, pause]);
+
+  useEffect(() => {
+    if (gameOver) {
+      return;
+    }
+    if (start && !gameOver && !pause) {
+      const intervalId = setTimeout(() => {
```

Comment:
Naming - using interval for `setTimeout` looks weird. Probably you just copy the name from the 53rd line. Check before pushing.